src/main/python/pgshovel/streams/utilities.py

```diff
@@ -24,3 +36,43 @@ def __str__(self):
             active=','.join(map(str, active[:self.max])),
             truncated='' if len(active) <= self.max else ',+{0}...'.format(len(active) - self.max),
         )
+
+
+class UnableToPrimeError(Exception):
+    """
+    Raised when an attempt to prime is made, but the priming failed after the
+    max number of messages.
+    """
+    pass
+
+
+def is_start_batch_operation(message):
+    """
+    Is the ``message`` the start operation of a batch of mutations.
+    """
+    value = get_oneof_value(message.batch_operation, 'operation')
+    return type(value) in TRANSACTION_START_EVENT_TYPES
+
+
+def prime_until(check, max_messages, stream):
+    """
+    Attempts to "prime" the ``stream`` by dropping any messages for which
+    ``check`` evaluates to ``False``, up until ``max_messages``.
+    """
+
+    attempts = count(start=1)
```

Comment:
I think you want this to be `start=0`, because as is, 
```
list(prime_until(lambda x: x!=0, max_messages=1, stream=[(0,0), (0,1),(0,2)])))
```
will raise UnableToPrimeError, whereas the intent I think is to drop the first message.